fathom/src/surface/elaboration.rs

```diff
@@ -981,14 +983,20 @@ impl<'interner, 'arena, 'error> Context<'interner, 'arena, 'error> {
     }
 
     /// Elaborate a module
-    pub fn elab_module(&mut self, surface_module: &Module<'_, ByteRange>) -> core::Module<'arena> {
+    pub fn elab_module(
+        &mut self,
+        surface_module: &Module<'_, ByteRange>,
+    ) -> Result<core::Module<'arena>, Vec<ModuleOrderMessage>> {
```

Comment:
I wonder if we could also add the messages to the elaboration context, to avoid needing to deal with them separately in the driver.

Reply:
Oh of course, this makes a lot of sense now that it's been moved into `elab_module`.

Reply:
I've updated the PR again with this change. Thanks again for the suggestions. It seems to have worked out pretty cleanly.